tests/integration/functional/test_client.py

```diff
@@ -1128,6 +1129,17 @@ def test_basic_crud_for_entity(
             pass
 
 
+class TestArtifact:
+    def test_prune(self, clean_client: "Client"):
```

Comment:
can you test that the two args for the function (deleting only artifacts and only_metadata) are also doing what they're meant to do. this test just tests it works generically, but we should test the edge cases too.

Reply:
add tests for that